src/libraries/System.Net.WebSockets.Client/tests/ConnectTest.Http2.cs

```diff
@@ -19,19 +19,29 @@ public class ConnectTest_Http2 : ClientWebSocketTestBase
     {
         public ConnectTest_Http2(ITestOutputHelper output) : base(output) { }
 
-        [Fact]
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
         [SkipOnPlatform(TestPlatforms.Browser, "Self-signed certificates are not supported on browser")]
-        public async Task ConnectAsync_VersionNotSupported_Throws()
+        public async Task ConnectAsync_VersionNotSupported_Throws(bool useHandler)
         {
             await Http2LoopbackServer.CreateClientAndServerAsync(async uri =>
             {
-                using (var clientSocket = new ClientWebSocket())
+                using (var cws = new ClientWebSocket())
                 using (var cts = new CancellationTokenSource(TimeOutMilliseconds))
                 {
-                    clientSocket.Options.HttpVersion = HttpVersion.Version20;
-                    clientSocket.Options.HttpVersionPolicy = Http.HttpVersionPolicy.RequestVersionExact;
-                    using var handler = CreateSocketsHttpHandler(allowAllCertificates: true);
-                    Task t = clientSocket.ConnectAsync(uri, new HttpMessageInvoker(handler), cts.Token);
+                    cws.Options.HttpVersion = HttpVersion.Version20;
+                    cws.Options.HttpVersionPolicy = Http.HttpVersionPolicy.RequestVersionExact;
+                    Task t;
+                    if (useHandler)
+                    {
+                        using var handler = new SocketsHttpHandler();
```

Comment:
Correct me if I'm wrong, but I believe the handler will get disposed after exiting the `if` block on line 40 (because of the `using`). I don't think it is expected?